File: src/booking/schemas/order.py
import uuid
from datetime import datetime
from decimal import Decimal

from pydantic import BaseModel, Field


class OrderItemRequest(BaseModel):
    ticket_type_id: uuid.UUID
    quantity: int = Field(ge=1, le=20)


class OrderCreateRequest(BaseModel):
    event_id: uuid.UUID
    items: list[OrderItemRequest] = Field(min_length=1, max_length=50)


class TicketRead(BaseModel):
    id: uuid.UUID
    ticket_type_id: uuid.UUID
    price: Decimal
    status: str


class OrderRead(BaseModel):
    id: uuid.UUID
    event_id: uuid.UUID
    status: str
    total_amount: Decimal
    reserved_until: datetime | None
    created_at: datetime
    tickets: list[TicketRead]


class OrderListResponse(BaseModel):
    items: list[OrderRead]
    total: int

File: frontend/src/types/order.ts
export type OrderStatus = 'reserved' | 'paid' | 'cancelled' | 'refunded';

export interface TicketRead {
  id: string;
  ticket_type_id: string;
  price: string;
  status: string;
}

export interface OrderRead {
  id: string;
  event_id: string;
  client_id?: string | null;
  status: OrderStatus;
  total_amount: string;
  reserved_until: string | null;
  created_at: string;
  tickets: TicketRead[];
}

export interface OrderItemRequest {
  ticket_type_id: string;
  quantity: number;
}

export interface OrderCreateRequest {
  event_id: string;
  items: OrderItemRequest[];
}
